pkg/webhook/handler/mutate/sidecar.go

```diff
@@ -0,0 +1,197 @@
+/*
+ Copyright 2022 Juicedata Inc
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package mutate
+
+import (
+	"context"
+	"encoding/json"
+	"path/filepath"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog"
+
+	"github.com/juicedata/juicefs-csi-driver/pkg/config"
+	"github.com/juicedata/juicefs-csi-driver/pkg/juicefs"
+	"github.com/juicedata/juicefs-csi-driver/pkg/juicefs/mount/builder"
+	"github.com/juicedata/juicefs-csi-driver/pkg/k8sclient"
+	"github.com/juicedata/juicefs-csi-driver/pkg/util"
+)
+
+type SidecarMutate struct {
+	Client  *k8sclient.K8sClient
+	juicefs juicefs.Interface
+
+	PVC        *corev1.PersistentVolumeClaim
+	PV         *corev1.PersistentVolume
+	jfsSetting *config.JfsSetting
+}
+
+var _ Mutate = &SidecarMutate{}
+
+func NewSidecarMutate(client *k8sclient.K8sClient, jfs juicefs.Interface, pvc *corev1.PersistentVolumeClaim, pv *corev1.PersistentVolume) Mutate {
+	return &SidecarMutate{
+		Client:  client,
+		juicefs: jfs,
+		PVC:     pvc,
+		PV:      pv,
+	}
+}
+
+func (s *SidecarMutate) Mutate(pod *corev1.Pod) (out *corev1.Pod, err error) {
```

Comment:
I think this method should pass context from caller, as webhook should timeout fastly, and the s.createOrUpdateSecret may cost much time.